### PR TITLE
feat(Checkbox): add checkbox component

### PR DESCRIPTION
**Pull request by brandonlilly**
**Before submitting a pull request,** please make sure the following is done:

I have done **all** of the following:

- [x] Added a top level class to all my components `'.anchor-[COMPONENT NAME]'`.
- [x] Used [conventional commits](https://www.conventionalcommits.org) for all work.
- [x] Tested my solution on Mobile & Tablet.
- [x] Wrote [unit tests](https://jestjs.io/docs/en/getting-started) for states and all behavior (`npm test`) and passed coverage thresholds.
- [x] Updated snapshots for all permutations (`npm test -- -u`).
- [x] Accounted for hover, focus, blur, visited, & error states because they are not edge cases.
- [x] Created TODOs for known edge cases.
- [ ] Documented all of my changes (inline & doc site).
- [x] Made sure that all accessibility errors are resolved.
- [x] Added [stories](https://storybook.js.org/docs/basics/introduction/) with knobs for all possible configurations.
- [x] De-linted and ran [prettier](https://github.com/prettier/prettier) (`npm run pretty`) on my code.
- [x] Added name to OWNERS file for all new components
- [x] If adding a new component, add its export to the rollup config

---------
**Outline your feature or bug-fix below**

Adds a Checkbox component

### CODE REVIEW

File: src/Form/Checkbox/Checkbox.component.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import * as React from 'react';
+const { forwardRef } = React;
+
+// VENDOR
+import styled, { css } from '@xstyled/styled-components';
+import { compose } from '@xstyled/system';
+import classnames from 'classnames';
+
+// ANCHOR
+import { Check } from '../..';
```
Comment (sharkparty) [inline src/Form/Checkbox/Checkbox.component.tsx]: Does this give you a circular reference error on build?

File: src/Form/Checkbox/Checkbox.component.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+const StyledCheckbox = styled('label')<CheckboxProps>`
+    box-sizing: border-box;
+    position: relative;
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+
+    user-select: none;
+    border: dark;
+    border-radius: 2px;
```
Comment (sharkparty) [inline src/Form/Checkbox/Checkbox.component.tsx:42]: I'm guessing we don't want to codify this as a `small`?

File: src/Form/Checkbox/Checkbox.component.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+`;
+
+interface CheckboxProps extends SpaceProps, ColorProps, BackgroundColorProps {
+    className?: string;
+    id?: string;
+    htmlFor?: string;
+
+    checked?: boolean;
+    disabled?: boolean;
+    name?: string;
```
Comment (sharkparty) [inline src/Form/Checkbox/Checkbox.component.tsx:74]: I don't know if this should be optional.

File: src/Form/Checkbox/Checkbox.component.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    ${StyledIcon} {
+        transition: opacity 100ms ease-in-out;
+        ${({ checked }) =>
+            css({
+                opacity: checked ? 1 : 0,
+            })}
+    }
+`;
+
+interface CheckboxProps extends SpaceProps, ColorProps, BackgroundColorProps {
```
Comment (sharkparty) [inline src/Form/Checkbox/Checkbox.component.tsx:67]: Given the event handlers, you may not be able to use `React.HTMLAttributes<HTMLInputElement>`, but can you verify that? I know AutoComplete has too many custom handlers to be compatible with that interface.

**Review by sharkparty — CHANGES_REQUESTED** [review]
Extremely minor feedback. This is mostly solid.

File: src/Form/Checkbox/Checkbox.component.tsx (hunk 2 above)
Comment (brandonlilly) [reply src/Form/Checkbox/Checkbox.component.tsx:42]: Maybe we want to. Not sure how snowflake it is

File: src/Form/Checkbox/Checkbox.component.tsx (hunk 1 above)
Comment (brandonlilly) [reply src/Form/Checkbox/Checkbox.component.tsx]: Woops

File: src/Form/Checkbox/Checkbox.component.tsx (hunk 2 above)
Comment (sharkparty) [reply src/Form/Checkbox/Checkbox.component.tsx:42]: Let's leave this as is, but can you add a TODO calling this out?

File: src/Form/Checkbox/Checkbox.component.tsx (hunk 4 above)
Comment (brandonlilly) [reply src/Form/Checkbox/Checkbox.component.tsx:67]: Wasn't compatible, but how about:
```
    inputProps?: React.HTMLAttributes<HTMLInputElement>;
```
?

Comment (sharkparty) [reply src/Form/Checkbox/Checkbox.component.tsx:67]: This won't allow event binding on the parent checkbox, but it does solve a part of the issue. It's ok if this is omitted for now. I need to figure out a FormControl interface that allows us more flexibility.

File: src/Form/Checkbox/Checkbox.component.tsx (hunk 3 above)
Comment (brandonlilly) [reply src/Form/Checkbox/Checkbox.component.tsx:74]: I started switching and noticed that I have `ControlLabel`s automatically propagate it's `name` down to the control you give it. So if `name` was required, you'd have to specify the `name` for both anyway.

**Review by sharkparty — APPROVED** [review]

**Pull request merged**